File: 프로그래머스/0/120826. 특정 문자 제거하기/특정 문자 제거하기.java
class Solution {
    public String solution(String my_string, String letter) {
        int count = 0;
        for(int i = 0; i < my_string.length(); i++) {
            if (my_string.charAt(i) == letter.charAt(0)) {
                count++;
            }
        }

        char[] ch = new char[my_string.length() - count];
        int j = 0;

        for(int i = 0; i < my_string.length(); i++) {
            if (my_string.charAt(i) != letter.charAt(0)) {
                ch[j] = my_string.charAt(i);
                j++;
            }
        }

        String answer = new String(ch);
        return answer;
    }
}

File: 프로그래머스/0/120826. 특정 문자 제거하기/특정 문자 제거하기.cs
using System;

public class Solution {
    public string solution(string my_string, string letter) {
        int count = 0;
        for(int i = 0 ; i < my_string.Length; i++){
            if(my_string[i] != letter[0]){
                count++;
            }
        }
        char[] ch = new char[count];
        int index = 0;
        for(int i = 0 ; i < my_string.Length; i++){
            if(my_string[i] != letter[0]){
            ch[index++] = my_string[i];
            }
        }
        string answer = string.Concat(ch);
        return answer;
    }
}